test/cmp_hdr_test.c

```diff
@@ -161,22 +161,32 @@ static int test_HDR_set1_recipient(void)
 
 static int execute_HDR_update_messageTime_test(CMP_HDR_TEST_FIXTURE *fixture)
 {
-    struct tm hdrtm;
+    struct tm hdrtm, tmptm;
     time_t hdrtime, before, after, now;
 
     now = time(NULL);
-    before = mktime(gmtime(&now));
+    /*
+     * Trial and error reveals that passing the return value from gmtime
+     * directly to mktime in a mingw 32 bit build gives unexpected results. To
+     * work around this we take a copy of the return value first.
```

Comment:
I wonder if this is related to gmtime() using a static buffer which is overwritten.

Reply:
It's unclear, but as a single threaded app this shouldn't make a difference.

Reply:
More along the lines of something in mktime overwriting the gmtime buffer.

Reply:
That's plausible.

Reply:
Seems likely. Microsoft's documentation even notes:
> Note that gmtime and localtime (and _gmtime32, _gmtime64, _localtime32, and _localtime64) use a single buffer per thread for the conversion. If you supply this buffer to mktime, _mktime32 or _mktime64, the previous contents are destroyed.

https://docs.microsoft.com/en-us/cpp/c-runtime-library/reference/mktime-mktime32-mktime64?view=msvc-160

While this doesn't say that you can't pass the buffer to `mktime` (which suggests it's allowed), it does say that the Windows `mktime` touches the buffer. Perhaps the Windows or Visual Studio `mktime` internally makes a copy to account for this, but the Wine or MinGW doesn't?